backend/server/services/questionService.js

```diff
@@ -0,0 +1,85 @@
+const questionRepository = require('../repositories/questionRepository');
+
+function missingInputsThrowsValidationError(title, complexity, description) {
+  const innerText = description.replace(/<[^>]+>|\s+/g, '');
+  if (!title || !complexity || !innerText) {
+    throw { status: 400, message: 'Missing inputs' };
+  }
+}
+
+async function duplicateTitleThrowsDuplicateError(id, title) {
+  // id == null then simply check if title exist, else cross-check id
+  const isDuplicateTitle = await questionRepository.findByTitle(title);
+  if (isDuplicateTitle) {
+    if (id == null) {
+      throw { status: 409, message: `Question title already exist` };
+    } 
+
+    const isCurrent = await questionRepository.findByIdAndTitle(id, title);
+    if (!isCurrent) {
+      throw { status: 409, message: `Question title already exist` };
+    }
```

Comment:
Personally, the logic was a bit hard to read. (_Arrowhead code style spotted!_). Perhaps you can change the logic to something like:
```
const isDuplicateTitle = await ...;
if (!isDuplicateTitle) return;
// Duplicate Found -> check if match id
if (id == null) throw {...};
...
```

Reply:
Resolved